assets/blocks/course-outline/edit.js

```diff
@@ -1,21 +1,44 @@
 import { InnerBlocks } from '@wordpress/block-editor';
 import { useSelect, withSelect } from '@wordpress/data';
-import { useEffect } from '@wordpress/element';
+import { createContext, useEffect } from '@wordpress/element';
 import { CourseOutlinePlaceholder } from './placeholder';
 import { COURSE_STORE } from './store';
 import { useBlocksCreator } from './use-block-creator';
+import { OutlineBlockSettings } from './settings';
+
+/**
+ * A React context which contains the attributes and the setAttributes callback of the Outline block.
+ */
+export const OutlineAttributesContext = createContext();
```

Comment:
I've generally avoided named/default export mixing, but not sure why. I'd be interested in hearing what other folks think.

Reply:
If we avoid mixing it would make the defalut export a bit limiting since we won't be able to export anything else from the file. But I would also like to hear other people's ideas too, since I can't see any benefits with default exports other than avoiding to define an extra variable.

Reply:
I think it's fine, and it's pretty common in Gutenberg, exporting the main component as the default and other stuff as named.

Reply:
Cool! That works for me.